mgmt/ProcessManager.cc

```diff
@@ -176,6 +174,7 @@ ProcessManager::processManagerThread(void *arg)
     }
   }
 
+  pmgmt->processSignalQueue(); // ensure we're not leaking any messages
```

Comment:
This seems risky to me, since in many cases the loop will exit precisely when `pmgmt` isn't running. It's also possible to miss write errors. Or, on the other hand, if that can't happen no need to bother the final queue processing. I have to wonder, what exactly could be putting stuff on this queue if not the LM? If `pmgmt->initLMConnection()` doesn't get called, can the queue ever be non-empty? Maybe that should have an `else` that just returns from the method.

Reply:
You're right. It shouldn't be sending signals out at this point. I was missing this this precisely because I wasn't looking at the return values. It might be writing to a socket that is closed. Returning from the `else` is a better solution.